cleanlab/classification.py

```diff
@@ -512,31 +512,29 @@ class 0, 1, ..., K-1. `pred_probs` should have been computed using 3 (or
                 if self.verbose:
                     print("Fitting final model on the clean data ...")
             else:
-                if (
-                    "sample_weight" in inspect.getfullargspec(self.clf.fit).args
-                    and "sample_weight" not in self.clf_final_kwargs
-                    and self.noise_matrix is None
-                ):
-                    print(
-                        "Cannot utilize sample weights for final training. To utilize must either specify noise_matrix "
-                        "or have previously called self.find_label_issues() instead of filter.find_label_issues()"
-                    )
-
-                if self.verbose:  # pragma: no cover
+                if self.verbose:
                     if "sample_weight" in self.clf_final_kwargs:
-                        print("Fitting final model on the clean data with custom sample_weight...")
+                        print("Fitting final model on the clean data with custom sample_weight ...")
                     else:
+                        if (
+                            "sample_weight" in inspect.getfullargspec(self.clf.fit).args
+                            and self.noise_matrix is None
+                        ):
+                            print(
+                                "Cannot utilize sample weights for final training. To utilize must either specify noise_matrix "
```

Comment:
nit but This can be a bit clearer. 

i'd state it as 'Cannot use sample weights for final training because your model's fit function doesn't support the parameter 'sample_weight'.